applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -252,10 +267,22 @@ def InitializeSolutionStep(self):
 
             # Update the DENSITY and DYNAMIC_VISCOSITY values according to the new level-set
             self._SetNodalProperties()
-
+          
             # Initialize the solver current step
             self._GetSolutionStrategy().InitializeSolutionStep()
 
+            # Accumulative Water volume error ratio due to level Set .Adding source term 
+            water_volume_after_transport = KratosCFD.FluidAuxiliaryUtilities.CalculateFluidNegativeVolume(self.GetComputingModelPart())
+
+            volume_error = (water_volume_after_transport - system_volume) / system_volume
+            mass_source = False
+            if (self.settings["formulation"].Has("mass_source")):
+                mass_source = self.settings["formulation"]["mass_source"].GetBool()
```

Comment:
This has to be done once. No need to check if `mass_source` is active at each time step.

Reply:
True. It has been moved to Initialize instead InitializeSolutionStep